radicle-cob/src/object/collaboration.rs

```diff
@@ -53,3 +53,43 @@ impl CollaborativeObject {
         self.history.tips().into_iter().map(Oid::from).collect()
     }
 }
+
+/// Takes a `refname` and performs a best attempt to extract out the
+/// [`TypeName`] and [`ObjectId`] from it.
+///
+/// This assumes that the `refname` is in a
+/// [`git_ref_format::Qualified`] format. If it has any
+/// `refs/namespaces`, they will be stripped to access the underlying
+/// [`git_ref_format::Qualified`] format.
+///
+/// In the [`git_ref_format::Qualified`] format it assumes that the
+/// reference name is of the form:
+///
+///   `refs/<category>/<typename>/<object_id>[/<rest>*]`
+///
+/// Note that their may be more components to the path after the
+/// [`ObjectId`] but they are ignored.
+///
+/// Also note that this will return `None` if:
+///
+///   * The `refname` is not [`git_ref_format::Qualified`]
+///   * The parsing of the [`ObjectId`] fails
+///   * The parsing of the [`TypeName`] fails
+pub fn parse_refstr<R>(name: &R) -> Option<(TypeName, ObjectId)>
+where
+    R: AsRef<git_ref_format::RefStr>,
+{
+    use git_ref_format::Qualified;
+    let name = name.as_ref();
+    let refs_cobs = match name.to_namespaced() {
+        None => Qualified::from_refstr(name)?,
```

Comment:
Good to know the '?' can be used on options too :)

Reply:
Do you see this being used much elsewhere outside of storage/git.rs where there is an expectation?

It may be better to return an error, saving the caller the need to write a 'badly formatted [something]' here.

Reply:
This is a pretty low-level function, so I don't see it used in many places... I personally am fine with `Option` because I wouldn't handle the various errors differently, ie. it's either a valid cob path or it isn't; doesn't really matter which of the potential errors are returned.

Reply:
Ya, I'm inclined to leave it as `Option`. The function is well documented (and tested now) so if there's an issue we can refer to the docs :)